x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/EnrollmentToken.java

```diff
@@ -24,6 +24,10 @@
 import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
 
 public class EnrollmentToken {
+    // This was previously a version string, e.g. 8.12.0, but treated exclusively as a string everywhere, never parsed into a version.
+    // Arbitrarily set to 9 when decoupling this from node version.
+    public static final String CURRENT_TOKEN_VERSION = "9";
```

Comment:
Do we want to irrevocably decouple it from major version? There's still a concept of major version in the codebase, so it is possible we would still want to allow that boundary to be defined here?

Reply:
I'd be open to keeping it the same format, but it sort of seems like decoupled version numbers are the way to go these days anyway - I wouldn't be comfortable making assumptions about what any future major version means in a technical sense at this point.

Reply:
The API key version is being frozen at 8.13, we could do the same thing here?

We still don't have a good idea how strictly we want to enforce various boundaries around major versions, and how that is propagated into the various versions code. I have a slight preference for keeping with a 8.x prefix here, giving us space to do whatever we might need to do for ES 9.

Reply:
👍🏼 I'll switch it back to the old format.